compiler/rustc_mir_transform/src/coroutine.rs

```diff
@@ -317,6 +317,23 @@ impl<'tcx> TransformVisitor<'tcx> {
                         ),
                         user_ty: None,
                     })))
+                } else if matches!(
+                    val,
+                    Operand::Constant(box ConstOperand {
+                        const_: Const::Val(ConstValue::Scalar(_), _),
+                        ..
+                    })
+                ) {
+                    // yield (const false) is used for async drop yields inside AsyncGen
+                    // We need to convert value to Poll<OptRet>::Pending
```

Comment:
Can we directly generate the right yield in the async drop logic?

Reply:
TerminatorKind::Yield value is `Operand`, but we need `Rvalue::Aggregate` (`make_aggregate_adt`).

As far as I understand, we can do `Operand` with an additional local variable.
So, instead of 'return Poll<OptRet>::Pending;', the generated code will be 'x = Poll<OptRet>::Pending; return x;'

Value for Yield is generated here: https://github.com/rust-lang/rust/blob/b17dba45186c454576d0fc8fb93ecc65eb1a763a/compiler/rustc_mir_transform/src/coroutine/drop.rs#L386.

Does it cost additional temporary local?

Reply:
We can generate a const operand directly for a value of `Poll::Pending(())`, so that would not be the issue.